File: SP-API/Domain/Status.cs
using System.Collections.Generic;

namespace Domain
{
    public class Status
    {
        public int Id { get; set; }
        public int StatusTypeId { get; set; }
        public string Name { get; set; }
        public virtual StatusType StatusType { get; set; }
        public virtual ICollection<ItemBatch> ItemBatches { get; set; }
    }
}

File: SP-Client/src/app/_models/status.ts
export class Status {
  id: number;
  name: string;
  statusTypeName: string;
  statusTypeId: number;

  constructor(status) {
    this.id = status.id;
    this.name = status.name;
    this.statusTypeId = status.statusTypeId;
  }
}
